### PR TITLE
feat(mockController): handle onDidPresent in mockController

### PR DESCRIPTION
**Pull request by veiko**
Adds support for `onDidPresent` to `mockController`

### CODE REVIEW

File: src/mocks/mockController.tsx [hunk 1]
```diff
@@ -18,12 +18,13 @@ const MockControllerInner: React.FC<{
   onDidPresent?: () => void;
   onWillDismiss?: () => void;
   onWillPresent?: () => void;
-}> = ({ children, isOpen, header = '', buttons = [], onDidDismiss, message, forwardedRef, ...rest }) => {
+}> = ({ children, isOpen, header = '', buttons = [], onDidDismiss, onDidPresent, message, forwardedRef, ...rest }) => {
```
Comment (eric-horodyski) [inline src/mocks/mockController.tsx:21]: Would you mind adding `onWillDismiss` and `onWillPresent` to the list of parameters here?

File: src/mocks/mockController.tsx [hunk 2]
```diff
@@ -18,12 +18,13 @@ const MockControllerInner: React.FC<{
...
   const [open, setOpen] = useState(isOpen);
 
   useEffect(() => {
     if (isOpen && !open) {
       setOpen(true);
+      onDidPresent && onDidPresent();
```
Comment (eric-horodyski) [inline src/mocks/mockController.tsx:27]: Above this line, can we additionally add `onWillPresent && onWillPresent();`?

File: src/mocks/mockController.tsx [hunk 3]
```diff
@@ -18,12 +18,13 @@ const MockControllerInner: React.FC<{
...
     } else if (!isOpen && open) {
       setOpen(false);
```
Comment (eric-horodyski) [inline src/mocks/mockController.tsx:29]: Below this line, can we additionally add `onWillDismiss && onWillDismiss();`?

**Review by eric-horodyski — COMMENTED** [review]
Hi @veiko, thanks for the Pull Request! There's an additional issue (#7) that can be resolved alongside this PR. I've gone ahead and placed comments along with the code adjustments needed to resolve both the issue you've created this Pull Request to solve, as well as the additional issue linked above. Would you mind making the additional updates noted in the comments?

**Comment by veiko** [comment]
@eric-horodyski Added support for `onWillDismiss` and `onWillPresent` as well. Thanks for the review!

**Review by elylucas — APPROVED** [review]

**Comment by elylucas** [comment]
Thanks @veiko!

**Comment by veiko** [comment]
@elylucas Is there something I need to do for this to be released? Thanks!

**Pull request merged**